### PR TITLE
Upstream

### PR DESCRIPTION
**Pull request by Titaniumtown**
Includes AirplaneLite's dynamic activation range BUT I removed the villager portion because that can be handled in `purpur.yml`

### CODE REVIEW

**Review by duplexsystem — CHANGES_REQUESTED** [review]
I think we should merge purpur and airplane stuff or drop purpur stuff

File: patches/AirplaneLite/patches/server/0012-Dynamic-activation-range.patch [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+@@ -43,10 +43,17 @@ public class PathfinderGoalSelector {
+     }
+ 
+     // Paper start
++    /*
+     public boolean inactiveTick() {
+         incRate();
+         return getCurRate() % getTickRate() == 0;
+     }
++    */
```
Comment (ishland) [inline patches/AirplaneLite/patches/server/0012-Dynamic-activation-range.patch]: Breaks compatibility

File: patches/AirplaneLite/patches/server/0012-Dynamic-activation-range.patch [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
++                        activateChunkEntities( chunk, player.getPositionVector() ); // AirplaneL
+                     }
+                 }
+             }
+@@ -244,7 +247,7 @@ public class ActivationRange
+      *
+      * @param chunk
+      */
+-    private static void activateChunkEntities(Chunk chunk)
++    private static void activateChunkEntities(Chunk chunk, Vec3D playerVec) // AirplaneL - add player location
```
Comment (ishland) [inline patches/AirplaneLite/patches/server/0012-Dynamic-activation-range.patch]: Breaks compatibility

**Review by ishland — CHANGES_REQUESTED** [review]

**Comment by Titaniumtown** [comment]
Ok, I removed the DEAR patch in favor of Purpur's patch, we can further look into DEAR later.

**Review by ishland — DISMISSED** [review]

**Review by duplexsystem — DISMISSED** [review]

**Review by ishland — APPROVED** [review]

**Review by duplexsystem — APPROVED** [review]

**Review by duplexsystem — APPROVED** [review]

**Pull request merged**